packages/java/hilla-plugin-base/src/main/java/dev/hilla/plugin/base/InitFileExtractor.java

```diff
@@ -0,0 +1,165 @@
+                                    applicationPackage);
+                            // adjust path
+                            item = item.replace("org/vaadin/example",
+                                    applicationPath);
+                            // write file
+                            var path = projectDirectory.resolve(item);
+                            var parent = path.getParent();
+
+                            if (parent != null) {
+                                Files.createDirectories(parent);
+                            }
+
+                            LOGGER.info("Adding endpoint {}", item);
+                            Files.writeString(path, content);
+                        } else {
+                            LOGGER.info("Extracting {}", item);
+                            var path = projectDirectory.resolve(item);
+                            var parent = path.getParent();
+
+                            if (parent != null) {
+                                Files.createDirectories(parent);
+                            }
+
+                            Files.copy(zip.getInputStream(entry), path,
+                                    StandardCopyOption.REPLACE_EXISTING);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private enum Frameworks {
+        REACT(REACT_SKELETON, REACT_FILE_LIST), LIT(LIT_SKELETON,
+                LIT_FILE_LIST);
+
+        private final String skeletonUrl;
+        private final List<String> items;
+
+        Frameworks(String skeletonUrl, List<String> items) {
+            this.skeletonUrl = skeletonUrl;
+            this.items = items;
+        }
+
+        public String getSkeletonUrl() {
+            return skeletonUrl;
+        }
+
+        public List<String> getItems() {
+            return items;
+        }
+    }
+
+    private Frameworks detectFramework() throws IOException {
+        var buildFiles = List.of("pom.xml", "build.gradle", "build.gradle.kts");
+        var buildFile = buildFiles.stream()
+                .filter(file -> Files.exists(projectDirectory.resolve(file)))
+                .findFirst().orElseThrow(() -> new IllegalArgumentException(
+                        "No build file found"));
+
+        // search `hilla-react-spring-boot-starter` or
+        // `hilla-spring-boot-starter` in build file
+        var content = Files.readString(projectDirectory.resolve(buildFile));
+        if (content.contains("hilla-react-spring-boot-starter")) {
+            return Frameworks.REACT;
+        } else if (content.contains("hilla-spring-boot-starter")) {
+            return Frameworks.LIT;
+        } else {
+            throw new IllegalArgumentException("No hilla starter found");
+        }
+    }
+
+    private String findSpringBootApplicationPackage() throws IOException {
+        return Files.walk(projectDirectory)
+                .filter(path -> path.toString().endsWith(".java"))
```

Comment:
Could we change the logic here so that instead of looking into all directories, we just look into `[project-root]/src` and then remove the condition for paths ending in ".java" for the future, for instance, to support Kotlin? Although, currently we don't fully support Kotlin out of the box. Just a suggestion.